storage/rocksdb/ha_rocksdb.cc

```diff
@@ -10439,31 +10517,23 @@ void ha_rocksdb::get_auto_increment(ulonglong off, ulonglong inc,
         //      may not be in our sequence, but it is guaranteed to be equal
         //      to or larger than any other value already inserted.
         //
-        //  For now I'm going to take option @2.
-        new_val = std::numeric_limits<longlong>::max();
+        //  For now I'm going to take option 2.
+        new_val = std::numeric_limits<ulonglong>::max();
         auto_incr = new_val;  // Store the largest value into auto_incr
```

Comment:
If you have `n * inc + off > ULLONG_MAX`, then you definitely have a UNSIGNED BIGINT here, so it's the same as max_val. I can add an assert if needed.

As a side note, a return value of ULLONG_MAX for `get_auto_increment` has special meaning. This will return `ER_AUTOINC_READ_FAILED` to the user. So we have a quirk due to the SE API that we have different autoinc behaviour between UNSIGNED BIGINT and any of the other integer type. For UNSIGNED BIGINT, you get `ER_AUTOINC_READ_FAILED` once you hit max, whereas for everyone else, you just get duplicate key error (or no error if autoinc is on SK).

Reply:
Yes, that makes sense. Can you update the diff with these comments on the logic of the math and what happens in the UNSIGNED BIGINT vs. non-UNSIGNED BIGINT cases?

Reply:
There's some interesting interaction with modify column. Could it lead to auto_incr being larger than max_val after modify is performed?

create table t1 (pk bigint(8) auto_increment, primary key(pk)) engine=rocksdb;
insert into t1 values (1), (2), (10), (130), (140);
delete from t1 where pk > 127;
alter table t1 modify column pk tinyint auto_increment;
show create table t1;

The current implementation seems to show auto_increment = 141.

Reply:
I just tested on this diff, it seems to also return 141. We'd probably need to hook into the alter code paths to update this info for the table.

Reply:
It gets corrected on the next insert though.

Reply:
Yeah, the modify column should work correctly with this patch. It does means we can't assert() though on the condition that the value is larger than max_val since it's possible.